openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -110,6 +111,23 @@
 # .. toggle_tickets: https://openedx.atlassian.net/browse/ARCHBOM-1718
 LOG_REQUEST_USER_CHANGES = getattr(settings, 'LOG_REQUEST_USER_CHANGES', False)
 
+# .. toggle_name: LOG_REQUEST_USER_CHANGE_HEADERS
+# .. toggle_implementation: SettingToggle
+# .. toggle_default: False
+# .. toggle_description: Turn this toggle on to log all request headers, for all requests, for all user ids involved in
+#      any user id change detected by safe sessions. The headers will provide additional debugging information. The
+#      headers will be logged for all requests up until LOG_REQUEST_USER_CHANGE_HEADERS_DURATION seconds after
+#      the time of the last mismatch. The header details will be encrypted, and only available with the private key.
+# .. toggle_warnings: To work correctly, LOG_REQUEST_USER_CHANGES must be enabled and ENFORCE_SAFE_SESSIONS must be
+#      disabled.
+# .. toggle_use_cases: opt_in
+# .. toggle_creation_date: 2021-12-22
+# .. toggle_tickets: https://openedx.atlassian.net/browse/ARCHBOM-1940
+LOG_REQUEST_USER_CHANGE_HEADERS = getattr(settings, 'LOG_REQUEST_USER_CHANGE_HEADERS', False)
```

Comment:
This is documented as a `SettingToggle` but draws directly from `settings`. I assume you actually want to use `SettingToggle` here?